modules/afuser/afuser.c

```diff
@@ -134,23 +144,24 @@ afuser_dd_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_options)
             line[0] = 0;
           strncpy(p, ut->ut_line, sizeof(line) - (p - line));
           msg_debug("Posting message to user terminal",
-#ifdef SYSLOG_NG_HAVE_MODERN_UTMP
-                    evt_tag_str("user", ut->ut_user),
-#else
-                    evt_tag_str("user", ut->ut_name),
-#endif
+                    evt_tag_str("user", _get_utmp_username(ut)),
                     evt_tag_str("line", line));
           fd = open(line, O_NOCTTY | O_APPEND | O_WRONLY | O_NONBLOCK);
-          if (fd != -1)
+          if (fd == -1)
+            {
+              msg_error("Opening tty device failed",
+                        evt_tag_str("user", _get_utmp_username(ut)),
+                        evt_tag_str("line", line),
+                        evt_tag_error("errno"));
+              continue;
```

Comment:
we might similarly want to disable the usertty() destination in this case, don't we?

Reply:
I've added the disabling part too. Thanks for the suggestion!